### PR TITLE
feat(notification): Add `NotificationWithActions` option

### PR DESCRIPTION
**Pull request by alexcarpenter**
🎟️ [Asana Task](https://app.asana.com/0/1200641486648086/1202077202881790)
🔍 [Preview Link](https://react-components-git-acnotification-with-actions-hashicorp.vercel.app/components/notification)

---

## Description

Adding a new notification type for consent manager redesign https://github.com/hashicorp/react-components/pull/590

CleanShot 2022-05-19 at 14 27 49@2x

### PR Checklist 🚀

Items in this checklist may not may not apply to your PR, but please consider each item carefully.

- [x] Add Asana and Preview links above.
- [x] Conduct thorough self-review.
- [x] Add or update tests as appropriate.
- [x] Conduct reasonable cross browser testing for both compatibility and responsive behavior (We have a [Sauce Labs](https://app.saucelabs.com/) account for this, if you don't have access, just ask!).
- [x] Conduct reasonable accessibility review (use the [WAS](https://accessible.org/Web-Accessibility-Standards-WAS-2.pdf) as a guide or an [axe browser plugin](https://www.deque.com/axe/) until we establish more formal checks).
- [x] Identify (in the description above) and document (add Asana tasks on [this board](https://app.asana.com/0/1100423001970639/list)) any technical debt that you're aware of, but are not addressing as part of this PR.

### CODE REVIEW

File: packages/notification/docs.mdx [hunk 1]
```diff
@@ -75,3 +75,22 @@ Notification component to be used within our [Toaster](/components/toast) compon
   onDismiss={() => {}}
 />
 `}</LiveComponent>
+
+## With actions
+
+<LiveComponent>{`
+<NotificationWithActions
+  heading="Notification heading"
```
Comment (pbortnick) [inline packages/notification/docs.mdx:83]: 🙋 Do we want to add `onDismiss={() => {}}` in here to show that this prop is also available in the `NotificationWithActions` component?

Comment (alexcarpenter) [reply packages/notification/docs.mdx:83]: Yeah, thats a good callout. Added https://github.com/hashicorp/react-components/pull/593/commits/2e302518dfe231a3050c33cea08800ab060dd5db

File: packages/notification/style.module.css [hunk 2]
```diff
@@ -114,6 +114,18 @@
...
 
+.heading {
+  --color: var(--wpl-neutral-700);
+
+  composes: g-type-body-strong from global;
+  margin: 0;
+  color: var(--color);
+
+  @nest .dark & {
+    color: var(--wpl-neutral-0);
```
Comment (EnMod) [inline packages/notification/style.module.css:125]: Very non-blocking ⛏️ : was there a specific motivation behind assigning a new value to `color` vs the `--color` custom property here? Curious!

Comment (alexcarpenter) [inline packages/notification/style.module.css:125]:
```suggestion
    --color: var(--wpl-neutral-0);
```

Comment (alexcarpenter) [reply packages/notification/style.module.css]: Good catch! updated.

**Review by EnMod — APPROVED** [review]
LGTM! Nice 🙌

**Pull request merged**